packages/pin-input/src/use-pin-input.ts

```diff
@@ -156,166 +170,173 @@ export function usePinInput(props: UsePinInputProps = {}) {
     [values, setValues, focusNext, onComplete, descendants.length],
   )
 
-  const clear = useCallback(() => {
+  const clear = React.useCallback(() => {
     const values: string[] = Array(descendants.length).fill("")
     setValues(values)
     const firstInput = descendants[0]
     firstInput.element?.focus()
   }, [descendants, setValues])
 
-  return {
-    id,
-    domContext,
-    setValue,
-    values,
-    setValues,
-    setMoveFocus,
-    clear,
-    onComplete,
-    placeholder,
-    manageFocus,
-    isDisabled,
-    isInvalid,
-  }
-}
+  const getNextValue = React.useCallback(
+    (value: string, eventValue: string) => {
+      let nextValue = eventValue
+      if (value?.length > 0) {
+        if (value[0] === eventValue.charAt(0)) {
+          nextValue = eventValue.charAt(1)
+        } else if (value[0] === eventValue.charAt(1)) {
+          nextValue = eventValue.charAt(0)
+        }
+      }
+      return nextValue
+    },
+    [],
+  )
 
-export type UsePinInputReturn = ReturnType<typeof usePinInput>
+  const [focusedIndex, setFocusedIndex] = React.useState(-1)
 
-export interface UsePinInputFieldProps {
-  ref?: Ref<HTMLInputElement>
-  onChange?: ChangeEventHandler
-  onKeyDown?: KeyboardEventHandler
-  onFocus?: FocusEventHandler
-  onBlur?: FocusEventHandler
-}
+  const getInputProps = React.useCallback(
```

Comment:
This is such a nice improvement. Moves `PinInput` closer to more recent component implementations.

Reply:
Exactly!